### PR TITLE
Integrated dependency manager

### LINKED ISSUE #29: Integrate dependency manager
We have already talked about that on Gitter, we want to include the dependencies and we want to manage the dependencies with `govendor`, so let's integrate it!

### PR DESCRIPTION
**Pull request by tboerger**
I have integrated all dependencies with govendor and also updated the travis configuration since we don't need to download the dependencies anymore. This also fixes #29.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,18 +7,14 @@ go:
...
 script: |
-  go build -v -tags "pam"
-
-  for pkg in $(go list ./... | grep -v /vendor/)
-  do
-    go test -v -race -cover -coverprofile $GOPATH/src/$pkg/coverage.out $pkg || exit 1
-  done
+  - go build -v -tags 'cert sqlite pam miniwinsvc'
+  - |
+    for pkg in $(go list ./... | grep -v /vendor/)
```
Comment (bkcsoft) [inline .travis.yml]: Doesn't govendor have something like `glide no-vendor` ?

File: .travis.yml [hunk 2]
```diff
@@ -7,18 +7,14 @@ go:
...
-  go build -v -tags "pam"
-
-  for pkg in $(go list ./... | grep -v /vendor/)
-  do
-    go test -v -race -cover -coverprofile $GOPATH/src/$pkg/coverage.out $pkg || exit 1
-  done
+script:
+  - go build -v -tags 'cert sqlite pam miniwinsvc'
+  - |
+    for pkg in $(go list ./... | grep -v /vendor/)
```
Comment (bkcsoft) [inline .travis.yml:14]: couldn't this be written as instead?

```
go test -v race -cover -coverprofile coverage.out $(go list ./... | grep -v /vendor/)
```

Or do we _need_ to split them up?

**Comment by andreynering** [comment]
You forgot to vendor test packages. See the Travis output:

```
models/models_test.go:10:2: cannot find package "github.com/smartystreets/goconvey/convey" in any of:
    /home/travis/gopath/src/github.com/go-gitea/gitea/vendor/github.com/smartystreets/goconvey/convey (vendor tree)
    /home/travis/.gimme/versions/go1.6.linux.amd64/src/github.com/smartystreets/goconvey/convey (from $GOROOT)
    /home/travis/gopath/src/github.com/smartystreets/goconvey/convey (from $GOPATH)
```

File: .travis.yml (hunk 2 above)
Comment (tboerger) [reply .travis.yml:14]: Nope, `cannot use test profile flag with multiple packages`

**Comment by tboerger** [comment]
> You forgot to vendor test packages. See the Trivis output

Working on it.

**Comment by tboerger** [comment]
Ready for review/merge

**Comment by bkcsoft** [comment]
LGTM

**Comment by tboerger** [comment]
Rebased

**Comment by lunny** [comment]
LGTM

**Pull request merged**